src/lib/downloads/downloadmanager.cpp

```diff
@@ -256,7 +256,7 @@ void DownloadManager::download(QWebEngineDownloadItem *downloadItem)
 
     // Create download item
     QListWidgetItem* listItem = new QListWidgetItem(ui->list);
-    DownloadItem* downItem = new DownloadItem(listItem, downloadItem, QFileInfo(downloadPath).absolutePath(), fileName, openFile, this);
+    DownloadItem* downItem = new DownloadItem(listItem, downloadItem, QFileInfo(downloadPath).absolutePath(), QFileInfo(downloadPath).fileName(), openFile, this);
```

Comment:
.fileName() extracts *correct* file name from path, whilst filename variable extracts file name from original URL. You can check that with qDebug().